jsettlers.logic/src/main/java/jsettlers/logic/map/grid/objects/MapObjectsManager.java

```diff
@@ -400,18 +401,14 @@ public boolean pushMaterial(short x, short y, EMaterialType materialType) {
 		}
 	}
 
-	public ShortPoint2D pushMaterialForced(short x, short y, EMaterialType materialType) {
-		IMapArea mapArea = new HexGridArea(x, y, 0, 200);
-
-		for (ShortPoint2D position : mapArea) {
-			short currX = position.x;
-			short currY = position.y;
+	public ShortPoint2D pushMaterialForced(int x, int y, EMaterialType materialType) {
+		return HexGridArea.iterate(x, y, 0, 200, (currX, currY, radius) -> {
```

Comment:
How do you mean? What should the JIT be able to inline and to where? 

My plan is to completely remove the iterators, as they create large amounts of objects. That's the main reason, why I replaced this code here. As this method is not called very often, speed doesn't matter that much, here. However, I don't want to have the iterators and the iterate methods in parallel, as they tend to have duplicated, complicated code.